### PR TITLE
portage: add v1.2.2

### PR DESCRIPTION
**Pull request by junghans**

### CODE REVIEW

File: var/spack/repos/builtin/packages/portage/package.py [hunk 1]
```diff
@@ -17,14 +17,19 @@ class Portage(CMakePackage):
 
     # tarballs don't have submodules, so use git tags
     version('develop', branch='master', submodules=True)
+    version('1.2.2', tag='v1.2.2', submodules=True)
     version('1.1.1', tag='v1.1.1', submodules=True)
     version('1.1.0', tag='v1.1.0', submodules=True)
 
+    patch('gcc-7.patch', when='@1.2.2 %gcc@7:')
+    patch('p_lapacke_config.patch', when='@1.2:')
```
Comment (citibeth) [inline var/spack/repos/builtin/packages/portage/package.py]: Why are these patches needed?  Why weren't they needed before?  Can you point to webpages explaining these issues in greater detail?  Is the `gcc-7` patch needed for `v1.1.0` and `v1.1.1`?

File: var/spack/repos/builtin/packages/portage/package.py [hunk 2]
```diff
@@ -17,14 +17,19 @@ class Portage(CMakePackage):
...
+
     variant('mpi', default=True, description='Support MPI')
 
     depends_on("cmake@3.1:", type='build')
     depends_on('mpi', when='+mpi')
     depends_on('lapack')
+    depends_on('boost')
```
Comment (citibeth) [inline var/spack/repos/builtin/packages/portage/package.py]: Why is this dependency needed but was not needed before?  (Or was it needed before and being picked up as a zombie dependency?)

Comment (junghans) [reply var/spack/repos/builtin/packages/portage/package.py]: Yeah, it usually was picked up from the system.

File: var/spack/repos/builtin/packages/portage/package.py (hunk 1 above)
Comment (junghans) [reply var/spack/repos/builtin/packages/portage/package.py]: I haven't tested all combinations, I just don't want to touch old versions.

The `gcc-7.patch` is needed as `fabs()` needs `math.h` with gcc-7, this issue got fixed in the code.

The `p_lapacke_config.patch` got fixed here: https://github.com/laristra/cinch/commit/f87f848269fac25aa5b8d0bd5d9c9b2d2d6fb0ad

Comment (citibeth) [reply var/spack/repos/builtin/packages/portage/package.py]: > The `p_lapacke_config.patch` got fixed here: [laristra/cinch@f87f848](https://github.com/laristra/cinch/commit/f87f848269fac25aa5b8d0bd5d9c9b2d2d6fb0ad)

Please add that URL into the package as a comment so people know why the patch is there.



> The `gcc-7.patch` is needed as `fabs()` needs `math.h` with gcc-7, this issue got fixed in the code.

What do you mean by "got fixed in the code"?  In the GCC code or the `portage` code?  If the latter, which version was it fixed in?  Please
1. Add a comment like the above, explaining why the patch is needed (and again; pointing to a URL if possible)
2. Change the `when=` clause to be your best guess as to the versions that need it.  If you think this is a GCC 7 issue that would most likely affect previous versions of `portage` (even if you haven't tested them), then that would be `when=%gcc@7:`.
3. Do you think that `1.2.2` is the last release of `portage` that needs this patch?  If so, then you would use `when=@:1.2.2 %gcc7%`.

Comment (junghans) [reply var/spack/repos/builtin/packages/portage/package.py]: Comments added.

File: var/spack/repos/builtin/packages/portage/package.py [hunk 3]
```diff
@@ -17,14 +17,22 @@ class Portage(CMakePackage):
 
     # tarballs don't have submodules, so use git tags
     version('develop', branch='master', submodules=True)
+    version('1.2.2', tag='v1.2.2', submodules=True)
     version('1.1.1', tag='v1.1.1', submodules=True)
     version('1.1.0', tag='v1.1.0', submodules=True)
 
+    # fabs() needs math.h for gcc-7, got fixed in version above 1.2.2
+    patch('gcc-7.patch', when='@1.2.2 %gcc@7:')
```
Comment (citibeth) [inline var/spack/repos/builtin/packages/portage/package.py]: Please change this to `when='@:1.2.2 %gcc@7:'` in order to match the comment.  I'm not worried that you haven't personally tested past versions.

Comment (junghans) [reply var/spack/repos/builtin/packages/portage/package.py]: I would prefer not touching and possibly breaking an old version by with unapplying patch, I will just update the comment.

**Comment by citibeth** [comment]
@ junghans

>     # fabs() needs math.h for gcc-7, got fixed in 
>	    # versions above 1.2.2, not tested for 1.1.*
>	    # but nobody complaint, so let's no touch it. 

Nobody has complained because nobody's tried building the old version with GCC7 yet.  I'd prefer we just fix it now, rather than leaving things we're almost certain are bugs in the system, and then only fixing them when someone complains.

**Comment by junghans** [comment]
Done.

**Pull request merged**